RHIDP-11676: Orchestrator - Migrate doc for Log aggregation and Open Telemetry feature (#2027)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/proc-connect-traces-to-existing-monitoring-to-avoid-tool-sprawl.adoc

:_mod-docs-content-type: PROCEDURE

[id="connect-traces-to-existing-monitoring-to-avoid-tool-sprawl_{context}"]
= Connect traces to existing monitoring to avoid tool sprawl

[role="_abstract"]
Route workflow traces to the observability tools your team already uses instead of learning a new system. Exporter configuration sends telemetry data to Jaeger, Grafana, or any OTLP-compatible platform.

.Prerequisites

* You have enabled OpenTelemetry in your workflow.
* An observability platform (Jaeger or OpenTelemetry Collector) is available in your cluster.

.Procedure

. Define your export strategy.
+

> Choose an export strategy that matches your observability platform requirements:

Reviewer: ```suggestion
. Configure an export strategy based on your environment requirements:
```

> ** OTLP exporter with batch processing (Recommended)

Reviewer: ```suggestion
** Configure the OTLP exporter with batch processing (Recommended)
```

+
For production environments, use an OTLP exporter with batch processing to reduce network overhead and improve performance:
+
[source,bash]
----
# OTLP exporter - Direct to Jaeger
quarkus.otel.exporter.otlp.endpoint=http://jaeger-collector.observability.svc.cluster.local:4317
quarkus.otel.exporter.otlp.protocol=grpc
quarkus.otel.traces.exporter=cdi

# Batch processing for production
quarkus.otel.bsp.schedule.delay=5s
quarkus.otel.bsp.max.export.batch.size=512
quarkus.otel.bsp.export.timeout=2s
quarkus.otel.bsp.max.queue.size=2048
----

> ** Direct export to an external platform

Reviewer: ```suggestion
** Configure direct export to an external platform
```

+
For development or simple integrations, use a direct export configuration:
+
[source,bash]
----
# Example: Direct export to Jaeger
quarkus.otel.exporter.otlp.endpoint=http://jaeger-collector:4317
quarkus.otel.exporter.otlp.protocol=grpc
quarkus.otel.traces.exporter=cdi
----

. Externalize the configuration for production deployments.
+

> Use environment variables to externalize your OpenTelemetry configuration. This ensures that your deployment remains secure and flexible across environments.

Reviewer: ```suggestion
. Externalize the configuration for production deployments by using environment variables. This ensures that your deployment remains secure and flexible across environments.
```
